tests/xharness/Jenkins.cs

```diff
@@ -629,7 +629,9 @@ async Task<IEnumerable<TestTask>> CreateRunDeviceTasksAsync ()
 				rv.AddRange (projectTasks);
 			}
 
-			return CreateTestVariations (rv, (buildTask, test, candidates) => new RunDeviceTask (buildTask, candidates?.Cast<Device> () ?? test.Candidates));
+			var tcs = new TaskCompletionSource<IEnumerable<TestTask>> ();
+			tcs.SetResult (CreateTestVariations (rv, (buildTask, test, candidates) => new RunDeviceTask (buildTask, candidates?.Cast<Device> () ?? test.Candidates)));
```

Comment:
TaskCompletionSource is not needed here. On synchronous code-paths, `Task.FromResult` can be used.